content/en/blog/_posts/2024-02-29-cri-o-seccomp-oci-artifacts.md

```diff
@@ -0,0 +1,421 @@
+    "size": 3
+  },
+  "layers": [
+    {
+      "mediaType": "application/vnd.oci.image.layer.v1.tar",
+      "digest": "sha256:c15f3eb1efd9abe0ae973e050c374988bba52c45a53f5c3cd80419a865013fa6",
+      "size": 20480,
+      "annotations": {
+        "org.opencontainers.image.title": "seccomp.tar"
+      }
+    }
+  ],
+  "annotations": {
+    "org.opencontainers.image.created": "2024-02-05T11:37:22Z"
+  }
+}
+```
+
+We can see that the image itself consists of an empty config
+(`application/vnd.oci.image.config.v1+json`) and a single layer
+(`application/vnd.oci.image.layer.v1.tar`) pointing to the `seccomp.tar` file.
+Let's give it a try!
+
+### Using the annotation for a specific container or whole pod
+
+CRI-O needs to be configured adequately before it can utilize the annotation. To
+do this, add the annotation to the `allowed_annotations` array for the runtime.
+This can be done by using a drop-in configuration
+`/etc/crio/crio.conf.d/10-crun.conf` like this:
+
+```toml
+[crio.runtime]
+default_runtime = "crun"
+
+[crio.runtime.runtimes.crun]
+allowed_annotations = [
+    "io.kubernetes.cri-o.seccompProfile",
+]
+```
+
+Now, let's run CRI-O from the latest `main` commit. This can be done by either
+building it from source, using the [static binary bundles](https://github.com/cri-o/packaging?tab=readme-ov-file#using-the-static-binary-bundles-directly)
+or [the prerelease packages](https://github.com/cri-o/packaging?tab=readme-ov-file#usage).
+
+For the demo I just ran the `crio` binary from my command line using a single
+node Kubernetes cluster via [`local-up-cluster.sh`](https://github.com/cri-o/cri-o?tab=readme-ov-file#running-kubernetes-with-cri-o).
+Now that the cluster is up and running, let's try a pod without the annotation
+running as seccomp `Unconfined`:
+
+```
+> cat pod.yaml
+```
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: pod
+spec:
+  containers:
+    - name: container
+      image: nginx:1.25.3
+      securityContext:
+        seccompProfile:
+          type: Unconfined
+```
+
+```
+> kubectl apply -f pod.yaml
+```
+
+The workload is up and running:
+
+```
+> kubectl get pods
+NAME   READY   STATUS    RESTARTS   AGE
+pod    1/1     Running   0          15s
+```
+
+And no seccomp profile got applied if we inspect the container using
+[`crictl`](https://sigs.k8s.io/cri-tools):
+
+```
+> export CONTAINER_ID=$(sudo crictl ps --name container -q)
+
+> sudo crictl inspect $CONTAINER_ID | jq .info.runtimeSpec.linux.seccomp
+null
+```
+
+Now, let's modify the pod to apply the profile `quay.io/crio/seccomp:v1` to the
+container:
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: pod
+  annotations:
+    io.kubernetes.cri-o.seccompProfile/container: quay.io/crio/seccomp:v1
+spec:
+  containers:
+    - name: container
+      image: nginx:1.25.3
+```
+
+We have to delete the workload, because only recreation will apply a new seccomp
```

Comment:
```suggestion
I have to delete and recreate the Pod, because only recreation will apply a new seccomp
```

We [avoid using “we”](https://kubernetes.io/docs/contribute/style/style-guide/#avoid-using-we); there's an exception for using it to refer to the multiple authors of a blog article, but that doesn't apply here.